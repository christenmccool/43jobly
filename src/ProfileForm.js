import React, {useState, useContext} from 'react';
import UserContext from './UserContext';
import './Form.css';

const ProfileForm = ({editProfile}) => {
  const user = useContext(UserContext);

  const initialFormData = {...user, password: ""};
  const [formData, setFormData] = useState(initialFormData);

  const handleChange = (evt) => {
    const {name, value} = evt.target;
    setFormData({...formData, [name]: value});
  }

  const handleSubmit = (evt) => {
    evt.preventDefault();
    editProfile(formData.username, 
                  {password: formData.password, 
                    firstName: formData.firstName, 
                    lastName: formData.lastName, 
                    email: formData.email
                  }
                );
    setFormData(initialFormData);
  }

  return (
    <div className="Form">
      <h1 className="Form-title">Profile</h1>
      <form className="Form-form" onSubmit={handleSubmit}>
        <label htmlFor="username">Username</label>
        <input 
          id="username" 
          name="username" 
          type="text" 
          value={user.username}
          readOnly
          disabled
        />
        <label htmlFor="firstName">First name</label>
        <input 
          id="firstName" 
          name="firstName" 
          type="text" 
          placeholder={user.firstName} 
          value={formData.firstName}
          onChange={handleChange}
          required
        />
        <label htmlFor="lastName">Last name</label>
        <input 
          id="lastName" 
          name="lastName" 
          type="text" 
          placeholder={user.lastName} 
          value={formData.lastName}
          onChange={handleChange}
          required
        />
        <label htmlFor="email">Email</label>
        <input 
          id="email" 
          name="email" 
          type="email" 
          placeholder={user.email} 
          value={formData.email}
          onChange={handleChange}
          required
        />
        <label htmlFor="password">Confirm password to make changes:</label>
        <input 
          id="password" 
          name="password" 
          type="password" 
          value={formData.password}
          onChange={handleChange}
          required
        />
        <button>Save Changes</button>
      </form>
    </div>
  )
}

export default ProfileForm;