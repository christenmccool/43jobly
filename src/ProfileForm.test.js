import { render, asFragment } from '@testing-library/react';
import ProfileForm from './LoginForm';

test('if it renders without crashing', () => {
  render(<ProfileForm />);
});

test('if it matches snapshot', () => {
  const {asFragment} = render(<ProfileForm />);
  expect(asFragment()).toMatchSnapshot();
});