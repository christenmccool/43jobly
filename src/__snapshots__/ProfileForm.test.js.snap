// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Form"
  >
    <h1
      class="Form-title"
    >
      Profile
    </h1>
    <form
      class="Form-form"
    >
      <label
        for="username"
      >
        Username
      </label>
      <input
        id="username"
        name="username"
        required=""
        type="text"
        value=""
      />
      <label
        for="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        required=""
        type="password"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
